### PR TITLE
Prepare for v3.24.0 release

### PR DESCRIPTION
**Pull request by fairclothjm**
Steps

- update Vault versions for the Acceptance test job test matrix
- update dependencies
- update CHANGELOG

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +10,27 @@ BUGS:
 * fix `vault_kv_secret_v2` drift when "data" is in secret name/path ([#2104](https://github.com/hashicorp/terraform-provider-vault/pull/2104))
 * fix `vault_database_secret_backend_connection`: allow mysql_rds,mysql_aurora,mysql_legacy options of vault_database_secret_backend_connection terraform resource to allow specifying tls_ca and tls_certificate_key ([#2106](https://github.com/hashicorp/terraform-provider-vault/pull/2106))
 
+IMPROVEMENTS:
```
Comment (vinay-gopalan) [inline CHANGELOG.md:13]: nit: I think we're missing the entries for 2 PRs:

- #2057 — looks like the changelog entry for this accidentally got added to the section for [v3.22.0](https://github.com/hashicorp/terraform-provider-vault/pull/2057/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR15)
- #2101 — missed the changelog entry 😅 would be great to add that and mention that the Azure Role changes are only for 1.16+

Comment (fairclothjm) [reply CHANGELOG.md:13]: nice catches! thanks

Comment (fairclothjm) [reply CHANGELOG.md:15]: done, amended and force pushed so we have a clean release commit :)

Comment (vinay-gopalan) [reply CHANGELOG.md:15]: Nice!

**Review by vinay-gopalan — APPROVED** [review]
LGTM!

**Pull request merged**